routing/payment_lifecycle.go

```diff
@@ -382,14 +385,15 @@ func (p *shardHandler) launchShard(rt *route.Route) (*channeldb.HTLCAttemptInfo,
 	if sendErr != nil {
 		// TODO(joostjager): Distinguish unexpected internal errors
 		// from real send errors.
-		err := p.failAttempt(attempt, sendErr)
+		htlcAttempt, err := p.failAttempt(attempt, sendErr)
 		if err != nil {
 			return nil, nil, err
 		}
 
 		// Return a launchOutcome indicating the shard failed.
 		return attempt, &launchOutcome{
-			err: sendErr,
+			attempt: htlcAttempt,
+			err:     sendErr,
```

Comment:
Not totally happy with this

Reply:
Can't we just use the `attempt` variable here as is?

Reply:
There isn't the same info in `attempt`. The `FailInfo` is missing.